### DISCUSSION TITLE
view_pandas accepting multiple DataFrame objects?

### DISCUSSION THREAD

**Discussion by jgarte** [saulpw/visidata#2258] [Ideas]
Hi,

What do you think if `view_pandas` could take any number of DataFrame objects?

Maybe it can open each one as a separate sheet?

**Comment by saulpw** [comment]
What would be the names of the sheets?

**Reply by jgarte** [reply]
```python
view_pandas(foo, bar, baz)
```

The sheets could be called `foo`, `bar`, and `baz` or `Foo`, `Bar`, and `Baz`, respectively?

**Reply by saulpw** [reply]
That would be great, but we can't reasonably get at those variable names from inside the function.  What about `view_pandas(foo=foo, bar=bar, baz=baz)`?

**Reply by jgarte** [reply]
That approach looks good to me :)

Should I create an issue for this feature request?

**Reply by saulpw** [reply]
The code is really simple, and would go into visidata/loaders/_pandas.py:

```
@VisiData.global_api
def view_pandas(vd, *args, **kwargs):
    sheets = [PandasSheet(k, source=v) for k, v in kwargs.items()]
    sheets.extend([PandasSheet('', source=v) for v in args])
    vd.run(*sheets)
```

The slightly harder part is testing it, adding a docstring, figuring out where else it should be documented, and submitting a pull request.  If you're willing then I'm sure it would much appreciated by other Pandas users!

**Reply by jgarte** [reply]
Nice! I'll try to take a look at those tasks this coming week/weekend.
